lib/css_parser/rule_set.rb

```diff
@@ -113,8 +117,62 @@ def delete(property)
       end
       alias remove_declaration! delete
 
-      def each(&block)
-        declarations.sort_by { |_name, value| value.order }.each(&block)
+      # Replace CSS property with multiple declarations
+      # @param [#to_s] property property name to be replaces
+      # @param [Hash<String => [String, Value]>] replacements hash with properties to replace with
+      #
+      # @example
+      #  declarations = Declarations.new('line-height' => '0.25px', 'font' => 'small-caps', 'font-size' => '12em')
+      #  declarations.replace_declaration!('font', {'line-height' => '1px', 'font-variant' => 'small-caps', 'font-size' => '24px'})
+      #  declarations
+      #  => #<CssParser::RuleSet::Declarations:0x00000000029c3018
+      #  @declarations=
+      #  {"line-height"=>#<CssParser::RuleSet::Declarations::Value:0x00000000038ac458 @important=false, @value="1px">,
+      #   "font-variant"=>#<CssParser::RuleSet::Declarations::Value:0x00000000039b3ec8 @important=false, @value="small-caps">,
+      #   "font-size"=>#<CssParser::RuleSet::Declarations::Value:0x00000000029c2c80 @important=false, @value="12em">}>
+      def replace_declaration!(property, replacements, preserve_importance: false)
+        property = normalize_property(property)
+        raise ArgumentError, "property #{property} does not exist" unless key?(property)
+
+        replacement_declarations = self.class.new(replacements)
+
+        if preserve_importance
+          importance = get_value(property).important
+          replacement_declarations.each { |_key, value| value.important = importance }
+        end
+
+        replacement_keys = declarations.keys
+        replacement_values = declarations.values
+        property_index = replacement_keys.index(property)
+
+        # We should preserve subsequent declarations of the same properties
+        # and prior important ones if replacement one is not important
+        replacements = replacement_declarations.each.with_object({}) do |(key, value), result|
+          if !declarations.key?(key) || (value.important && !declarations[key].important)
+            result[key] = value
+            next unless declarations.key?(key)
+
+            replaced_index = replacement_keys.index(key)
+            replacement_keys.delete_at(replaced_index)
+            replacement_values.delete_at(replaced_index)
+            property_index -= 1 if replaced_index < property_index
+            next
+          end
```

Comment:
```suggestion
          if !declarations.key?(key)
            result[key] = value
          elsif  !declarations[key].important && value.important # explain what this means
            result[key] = value
            replaced_index = replacement_keys.index(key)
            replacement_keys.delete_at(replaced_index)
            replacement_values.delete_at(replaced_index)
            property_index -= 1 if replaced_index < property_index
          elsif declarations[key].important && value.important # explain what this means
            # nothing to do ... explain why
          elsif property_index < replacement_keys.index(key)
            # nothing to do ... explain why
          else
            result[key] = value
          end
```

Reply:
Added more comments to the existing code.

Reply:
great comments!